nlopt_config.h.in

```diff
@@ -150,6 +150,9 @@
 /* Define if compiled including C++-based routines */
 #cmakedefine NLOPT_CXX
 
+/* Define if compiled including C++11-based routines */
+#cmakedefine NLOPT_CXX11
```

Comment:
Is this actually used anywhere?  Why do we need `-std=c++11` at all?

Reply:
The STOGO code is ancient — mostly written in 1998, and incorporated in NLopt in 2007.  How could it possibly depend heavily on C++11?

Reply:
I meant future code and the code from #194.